client/components/data/activating-theme/index.js

```diff
@@ -1,49 +1,49 @@
 /**
  * External dependencies
  */
-var React = require( 'react' );
+import React from 'react';
+import { connect } from 'react-redux';
+import omit from 'lodash/object/omit';
 
 /**
  * Internal dependencies
  */
-var CurrentThemeStore = require( 'lib/themes/stores/current-theme' );
-
-function getState( props ) {
-	return {
-		isActivating: CurrentThemeStore.isActivating(),
-		hasActivated: CurrentThemeStore.hasActivated(),
-		currentTheme: CurrentThemeStore.getCurrentTheme( props.siteId )
-	};
-}
+import {
+	isActivating,
+	hasActivated,
+	getCurrentTheme
+} from 'lib/themes/selectors/current-theme';
 
 /**
  * Passes the activating state of themes to the supplied child component.
  */
-var ActivatingThemeData = React.createClass( {
+const ActivatingThemeData = React.createClass( {
 
 	propTypes: {
-		children: React.PropTypes.element.isRequired
-	},
-
-	getInitialState: function() {
-		return getState( this.props );
-	},
-
-	componentWillMount: function() {
-		CurrentThemeStore.on( 'change', this.onActivatingTheme );
-	},
-
-	componentWillUnmount: function() {
-		CurrentThemeStore.off( 'change', this.onActivatingTheme );
-	},
-
-	onActivatingTheme: function() {
-		this.setState( getState( this.props ) );
+		children: React.PropTypes.element.isRequired,
+		// The following props aren't part of this module's public interface,
+		// since they are provided by the `connect()` call that is wrapped around
+		// `ActivatingThemeData` before it is exported.
```

Comment:
> My first question would be: is this information really necessary?

I don't mind really either way -- @ehg and @seear opined that we should remove those props altogether. Somehow, I felt it was giving a somewhat incomplete picture if we did that, so I chose to add those comments. But yeah, scale that to a 100 connected components :-/

> If it is, there's a subsequent question: can we convey that information through code? For instance, some types could be attached to the plain component, and some others to the connected component

Hmm, how would we achieve the latter? I can seem to think only of ES6 class/propTypes-style, i.e. adding a `propTypes` member after it's connected...

> If we prefer not to use code, can the comment be reduced to something like `// @connect`, `// @redux`, etc.?

Sounds okay to me, as long as it doesn't collide with JSDoc statements.